### PR TITLE
[ci skip] improve obsoletion message on DiscordOverwriteBuilder

### PR DESCRIPTION
**Pull request by ExaInsanity**
clarifies the message and updates it to specify 4.4 as removal version, as discussed.

skips CI because it's a change not affecting any logic

### CODE REVIEW

File: DSharpPlus/Entities/Channel/Overwrite/DiscordOverwriteBuilder.cs [hunk 1]
```diff
@@ -72,7 +72,7 @@ public DiscordOverwriteBuilder(DiscordRole role)
         /// <summary>
         /// Creates a new Discord permission overwrite builder. This class can be used to construct permission overwrites for guild channels, used when creating channels.
         /// </summary>
-        [Obsolete("Will be removed in 5.0. Use specialized constructors instead", false)]
+        [Obsolete("The parameterless constructor will be removed in 4.4. Use the specialized constructors instead", false)]
```
Comment (VelvetThePanda) [inline DSharpPlus/Entities/Channel/Overwrite/DiscordOverwriteBuilder.cs:75]: This should likely say *parameterized* constructors, given the confusion that's spurred from the wording.

**Review by VelvetThePanda — COMMENTED** [review]
LGTM outside of comment.

File: DSharpPlus/Entities/Channel/Overwrite/DiscordOverwriteBuilder.cs (hunk 1 above)
Comment (ExaInsanity) [reply DSharpPlus/Entities/Channel/Overwrite/DiscordOverwriteBuilder.cs:75]: good idea, will update

**Comment by ExaInsanity** [comment]
updated the message

**Pull request merged**